src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceTransformer.java

```diff
@@ -155,6 +166,81 @@ public JavaSurfaceTransformer(
     return surfaceDocs;
   }
 
+  /**
+   * Generates a list of standalone sample view models for the the API, at most one for each calling
+   * form for each method, as specified in each method view model's sampleValueSetsModel (which
+   * ultimately derives from user-provided configuration). If no samples are configured, no sample
+   * files are produced.
+   *
+   * @param context the interface for whose methods the sample files will be generated
+   * @param apiFile the previously generated API file view model for whose methods sample files will
+   *     be generated. The view models returned for each method are modified clones of the view
+   *     models for the methods in apiFile.
+   * @return A list of view models, each one corresponding to a distinct method sample
+   */
+  private List<ViewModel> generateSampleFilesForApi(
+      InterfaceContext context, StaticLangFileView<StaticLangApiView> apiFile) {
+    List<ViewModel> sampleDocs = new ArrayList<>();
+
+    StaticLangApiView classView = apiFile.classView();
+
+    final StaticLangFileView.Builder<StaticLangApiView> sampleFileBuilder =
```

Comment:
Done. But curious: why is that? Just because of the  clutter? I tend to like clarifying expectations for the compiler (and human readers) where the language allows. (I happen to like using `const int *const ptr` in C++)